### PR TITLE
Charlesmchen/sae vs tests 

### PR DESCRIPTION
**Pull request by charlesmchen**
Some important changes here... I'll call them out in comments.

PTAL @michaelkirk

### CODE REVIEW

File: Signal.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -191,36 +191,25 @@
 		34F308A21ECB469700BB7697 /* OWSBezierPathView.m in Sources */ = {isa = PBXBuildFile; fileRef = 34F308A11ECB469700BB7697 /* OWSBezierPathView.m */; };
 		34FD93701E3BD43A00109093 /* OWSAnyTouchGestureRecognizer.m in Sources */ = {isa = PBXBuildFile; fileRef = 34FD936F1E3BD43A00109093 /* OWSAnyTouchGestureRecognizer.m */; };
 		4505C2BF1E648EA300CEBF41 /* ExperienceUpgrade.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4505C2BE1E648EA300CEBF41 /* ExperienceUpgrade.swift */; };
-		4505C2C01E648EA300CEBF41 /* ExperienceUpgrade.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4505C2BE1E648EA300CEBF41 /* ExperienceUpgrade.swift */; };
```
Comment (charlesmchen) [inline Signal.xcodeproj/project.pbxproj]: swift files should NOT be added to both the `Signal` and `SignalTests` targets.  Instead we should `@testable import Signal`.  Adding them to both targets causes two separate copies to be linked.  One side effect is that we'd instantiate two copies of our `shared` singletons!

File: SignalMessaging/environment/Environment.m [hunk 2]
```diff
@@ -43,6 +43,11 @@ + (void)setCurrent:(Environment *)environment
     sharedEnvironment = environment;
 }
 
++ (void)clearCurrentForTests
+{
+    sharedEnvironment = nil;
```
Comment (charlesmchen) [inline SignalMessaging/environment/Environment.m]: Alternately we could remove the new assert in the setter.

File: Signal/test/util/DisplayableTextFilterTest.swift [hunk 3]
```diff
@@ -82,7 +83,7 @@ class DisplayableTextTest: XCTestCase {
         XCTAssertTrue("❤️💔💌💕💞💓💗💖💘💝💟💜💛💚💙".containsOnlyEmoji)
         XCTAssertTrue("✋🏿💪🏿👐🏿🙌🏿👏🏿🙏🏿".containsOnlyEmoji)
         XCTAssertTrue("🚾🆒🆓🆕🆖🆗🆙🏧".containsOnlyEmoji)
-        XCTAssertTrue("0️⃣1️⃣2️⃣3️⃣4️⃣5️⃣6️⃣7️⃣8️⃣9️⃣🔟".containsOnlyEmoji)
+        XCTAssertFalse("0️⃣1️⃣2️⃣3️⃣4️⃣5️⃣6️⃣7️⃣8️⃣9️⃣🔟".containsOnlyEmoji)
```
Comment (michaelkirk) [inline Signal/test/util/DisplayableTextFilterTest.swift]: wha? Did this change?

File: SignalMessaging/environment/Environment.m (hunk 2 above)
Comment (michaelkirk) [reply SignalMessaging/environment/Environment.m]: I prefer it as written for now. 👍

File: SignalMessaging/environment/Release.m [hunk 4]
```diff
@@ -14,11 +14,13 @@ @implementation Release
 
 + (Environment *)releaseEnvironment
 {
+    // Order matters here.
+    TSStorageManager *storageManager = [TSStorageManager sharedManager];
```
Comment (michaelkirk) [inline SignalMessaging/environment/Release.m]: Was something broken before? I can't think of any relevant side effects `[TSStorageManager sharedManager]` would have.

File: Signal/test/util/DisplayableTextFilterTest.swift (hunk 3 above)
Comment (charlesmchen) [reply Signal/test/util/DisplayableTextFilterTest.swift]: I think we removed this - due to feedback from you in an old PR - and I forgot to update the tests after taking your suggestion.

File: SignalMessaging/environment/Release.m (hunk 4 above)
Comment (charlesmchen) [reply SignalMessaging/environment/Release.m]: There was a race wherein (if I recall correctly) `OWSContactsManager`'s initializer kicks of `loadCachedDisplayNames` which async uses `TSStorageManager`, which expects to be initialized on the main thread.  Something like that.

**Pull request merged**